### PR TITLE
CircleCI Integration

### PR DESCRIPTION
**Pull request by asutula**

### CODE REVIEW

**Comment by asutula** [comment]
Ok, I'm ready to start writing some code again.

This PR makes it so every push to githib on any branch runs the JS, iOS, and Android tests. Any push to master will additionally create beta releases on Testflight and the Play Store.

I have NO IDEA why this PR lists all my commits (`asutula`) as by `textileio-machine` and all `textileio-machine`'s commits as `asutula`. WTF.

**Comment by asutula** [comment]
Failing iOS test due to a pbxproj error introduced by merging with master. Fixed now, should succeed soon

**Comment by asutula** [comment]
The circleci integration with Github PRs is sweet

File: android/app/build.gradle [hunk 1]
```diff
@@ -136,17 +120,17 @@ android {
...
+            keyAlias System.getenv("ANDROID_KEYSTORE_KEY_ALIAS")
+            keyPassword System.getenv("ANDROID_KEYSTORE_KEY_PASSWORD")
         }
     }
     buildTypes {
         release {
             minifyEnabled enableProguardInReleaseBuilds
             proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
-            if (file("./my-release-key.keystore").exists()) {
+            if (file(System.getenv("ANDROID_KEYSTORE")).exists()) {
```
Comment (sanderpick) [inline android/app/build.gradle]: sweet

**Comment by sanderpick** [comment]
damn, lots of work in there!

**Pull request merged**